packages/mongo/mongo_driver.js

```diff
@@ -1120,18 +1120,19 @@ MongoConnection.prototype.tail = function (cursorDescription, docCallback) {
   var cursor = self._createSynchronousCursor(cursorDescription);
 
   var stopped = false;
-  var lastTS = undefined;
+  var lastTS;
   var loop = function () {
+    var doc = null;
     while (true) {
       if (stopped)
         return;
       try {
-        var doc = cursor._nextObject();
+        doc = cursor._nextObject();
       } catch (err) {
         // There's no good way to figure out if this was actually an error
         // from Mongo. Ah well. But either way, we need to retry the cursor
         // (unless the failure was because the observe got stopped).
-        doc = null;
+        // `doc` is already `null` here;
```

Comment:
I don't think this change is safe or just a "linting" change – at least not without further consideration or explanation.

This is in a loop.  While `doc` is `null` the first time around, if `cursor._nextObject` throws an error, it would be left as the _previous_ `doc` that was processed on this cursor (loop).  A tailable cursor remains open until additional documents arrive.

Do you disagree?